### PR TITLE
docs: updating getting started

### PR DESCRIPTION
**Pull request by domgreen**
## Changes

Update so that its more explicit about adding cluster flags ... hopefully this means people can follow thru from the getting started, add cluster, expose store then add query.

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -47,22 +47,32 @@ If you are not interested in backing up any data, the `--gcs.bucket` flag can si
...
 * _[Example Kubernetes manifest with GCS upload](../kube/manifests/prometheus-gcs.yaml)_
 
-### Query Access
-
-Thanos comes with a highly efficient gRPC-based Store API for metric data access across all its components. The sidecar implements it in front of its connected Prometheus server. While it is ready to use with the above example, we must additionally configure the sidecar to join a Thanos cluster.
+### Gossip Cluster
 
 Components in a Thanos cluster are connected through a gossip protocol to advertise membership and propagate metadata about the data and services they provide.
 
 We a set of initial peers addresses to the sidecar. They only must be valid on startup and subsequent cluster changes are automatically adapted to.
```
Comment (Bplotka) [inline docs/getting_started.md]: "We a set" (not your) typo.

File: docs/getting_started.md [hunk 2]
```diff
@@ -47,22 +47,32 @@ If you are not interested in backing up any data, the `--gcs.bucket` flag can si
 * _[Example Kubernetes manifest](../kube/manifests/prometheus.yaml)_
 * _[Example Kubernetes manifest with GCS upload](../kube/manifests/prometheus-gcs.yaml)_
 
-### Query Access
```
Comment (Bplotka) [inline docs/getting_started.md]: I would stick to query access section first as it makes sense with the given flow
- backups
- query

File: docs/getting_started.md [hunk 3]
```diff
@@ -47,22 +47,32 @@ If you are not interested in backing up any data, the `--gcs.bucket` flag can si
...
-
-Thanos comes with a highly efficient gRPC-based Store API for metric data access across all its components. The sidecar implements it in front of its connected Prometheus server. While it is ready to use with the above example, we must additionally configure the sidecar to join a Thanos cluster.
+### Gossip Cluster
 
 Components in a Thanos cluster are connected through a gossip protocol to advertise membership and propagate metadata about the data and services they provide.
```
Comment (Bplotka) [inline docs/getting_started.md]: Here we can talk about how Query can talk to sidecars:
-  via gossip (and explain how)
- via static store flag (and explain how)

File: docs/getting_started.md [hunk 4]
```diff
@@ -71,6 +81,41 @@ Additional flags for cluster configuration exist but are typically not needed. C
...
 * _[Example Kubernetes manifest with GCS upload](../kube/manifests/prometheus-gcs.yaml)_
 
+### Query Access
+
+Thanos comes with a highly efficient gRPC-based Store API for metric data access across all its components. The sidecar implements it in front of its connected Prometheus server. While it is ready to use with the above example, we must additionally configure the sidecar to join a Thanos cluster.
+
+Let's extend the above sidecars to expose their gRPC Store API so that we can query metrics from both Prometheus instances.
+
+```
+thanos sidecar \
```
Comment (Bplotka) [inline docs/getting_started.md]: I don't think duplication will help here.. can we revert this and make sure gossip part is explained:
- WHY we need gossip
- how to use it (your modified sidecar examples makes sense just single sidecar + query would be just fine)
- What are alternatives (--store  flag)

File: docs/getting_started.md (hunk 2 above)
Comment (domgreen) [reply docs/getting_started.md]: Will move it after Query Access but before Query layer ... as i think its good to show how the query finds other components in the cluster. Thoughts?

Comment (Bplotka) [reply docs/getting_started.md]: perfect

File: docs/getting_started.md [hunk 5]
```diff
@@ -51,23 +51,20 @@ If you are not interested in backing up any data, the `--gcs.bucket` flag can si
 
 Thanos comes with a highly efficient gRPC-based Store API for metric data access across all its components. The sidecar implements it in front of its connected Prometheus server. While it is ready to use with the above example, we must additionally configure the sidecar to join a Thanos cluster.
 
-Components in a Thanos cluster are connected through a gossip protocol to advertise membership and propagate metadata about the data and services they provide.
-
-We a set of initial peers addresses to the sidecar. They only must be valid on startup and subsequent cluster changes are automatically adapted to.
+Let's extend the above sidecar to expose their gRPC Store API so that we can query metrics.
```
Comment (Bplotka) [inline docs/getting_started.md]: Can we add at line 49 something like: For details and configuration of other object stores see `./docs/storage.md`

File: docs/getting_started.md [hunk 6]
```diff
@@ -106,15 +109,83 @@ Providing the label name to the query component will enable the deduplication.
...
 
 Go to the configured HTTP address that should now show a UI similar to that of Prometheus itself. If the cluster formed correctly you can now query data across all Prometheus servers within the cluster.
 
 * _[Example Kubernetes manifest](../kube/manifests/thanos-query.yaml)_
 
+## Communication Between Components
+
+Components in a Thanos cluster can be connected through a gossip protocol to advertise membership and propagate metadata about other known nodes. We need to add gossip to efficiently discover other nodes in the cluster and the metrics information they can access.
+
+This is especially useful for the Query node to know all endpoints to query and the time windows of each nodes, thus reducing the overhead of querying all nodes in the cluster.
```
Comment (Bplotka) [inline docs/getting_started.md]: plus external labels of sidecars (if we are touching time windows)

File: docs/getting_started.md [hunk 7]
```diff
@@ -106,15 +109,83 @@ Providing the label name to the query component will enable the deduplication.
...
+```
+thanos query \
+    --http-address              0.0.0.0:19192 \         # Endpoint for Query UI
+    --grpc-address              0.0.0.0:19092 \         # gRPC endpoint for Store API
+    --cluster.address           0.0.0.0:19591 \
+    --cluster.advertise-address 127.0.0.1:19591 \
+    --cluster.peers             127.0.0.1:19391 \       # Static cluster peer where the node will get info about the cluster
+```
+
+The Query component however does not have to utilize gossip to discover other nodes and instead can be setup to use a static list of well known addresses to query. These are repeatable so can add as many endpoint as needed. However, if you only use `store` you will not be able to discover nodes added to the cluster.
```
Comment (Bplotka) [inline docs/getting_started.md]: ` if you only use `store` you will not be able to discover nodes added to the cluster.`

Can we put this into different words?  For example:
```
When to use gossip vs static stores?
- Use gossip if you want to maintain single gossip that is able to dynamically join and remove not visible components.
- Use static store when you want to have full control of what is connected to what. It is also easier to user static store options when setting up communication with remote (cross-cluster) components e.g (sidecar in different network through some proxy)
```

Comment (domgreen) [reply docs/getting_started.md]: updated wording and added the example below

**Pull request merged**